app/bundles/StageBundle/Tests/Unit/EventListener/CampaignSubscriberTest.php

```diff
@@ -101,6 +105,7 @@ public function __construct()
             {
             }
 
+            /** @phpstan-ignore-next-line */
```

Comment:
Same here. Is it the type of the param? Perhaps it could be fixed in the StageModel?

Reply:
I've added the comments. As it should be fixed in the StageModel and the AbstractCommonModel (parent). When I do it I get some PHPStan issues like
```
Parameter #1 $id of method Mautic\CoreBundle\Model\AbstractCommonModel<Mautic\ChannelBundle\Entity\MessageQueue>::getEntity() expects int|null, string given.
```

Do you think I should go ahead and try to solve the issues?

Reply:
That's what I usually do. I know it's annoying but it's the only way how to get the types under control. Once we'll have the right types then PHPSTAN and Rector can help us even more.

Reply:
@escopecz 

I found out that the `getEntity` method is called with 4 types of id: int|array<string, mixed>|string|null. The type hint can be used as is or simplified to `mixed` or `int|array<string, mixed>|null`. In the second case, I will convert `string` to `int` in multiple classes. 

Which way is better to use?

Reply:
I'd suggest to add this to the `AbstractCommonModel::getEntity()` method:
```php
/**
 * @param string|int|null $id
 */
```
I'd be curious to see which code is passing an array there. It wouldn't work well, would it?